modules/caddyhttp/fileserver/command.go

```diff
@@ -84,9 +86,34 @@ func cmdFileServer(fs caddycmd.Flags) (int, error) {
 	templates := fs.Bool("templates")
 	accessLog := fs.Bool("access-log")
 	debug := fs.Bool("debug")
+	compress := !fs.Bool("no-compress")
 
 	var handlers []json.RawMessage
 
+	if compress {
+		encodings := make(caddy.ModuleMap, 2)
+		prefer := make([]string, 0, 2)
+
+		gzip, err := caddy.GetModule("http.encoders.gzip")
+		if err != nil {
+			return 0, err
+		}
+		encodings["gzip"] = caddyconfig.JSON(gzip.New(), nil)
+		prefer = append(prefer, "gzip")
+
+		zstd, err := caddy.GetModule("http.encoders.zstd")
+		if err != nil {
+			return 0, err
+		}
+		encodings["zstd"] = caddyconfig.JSON(zstd.New(), nil)
+		prefer = append(prefer, "zstd")
```

Comment:
I think we usually put zstd before gzip because it _should_ perform better, but most clients don't support it (yet) so gzip is the fallback.

Reply:
Oops, it was my intent but I messed up the order in a refactoring. Fixed (and simplified) now.